yoyodyne/models/hard_attention.py

```diff
@@ -279,8 +285,12 @@ def greedy_decode(
             )
         return predictions, likelihood
 
+    # FIXME(kbg,bonham79): Needs input types and docs.
```

Comment:
Which is fixme, above or below?

Reply:
Below. Can you provide the types for that? And maybe some text for describing inputs and outputs?

Reply:
I'm not a fan of pushing to other's repos, so this would be the doc string:

```
    @staticmethod
    def greedy_step(
        log_probs: torch.Tensor, likelihood: torch.Tensor
    ) -> Tuple[torch.Tensor, torch.Tensor]:
        """Greedy decoding of current timestep.

        Args:
            log_probs (torch.Tensor): vocabulary probabilities at current time
                step.
            likelihood (torch.Tensor): accumulative likelihood of decoded character
                sequence.

        Returns:
            Tuple[torch.Tensor, torch.Tensors]: Greedy decoded character for current
                time step and current likelihood of decoded character sequence. 
        """
        tgt_prob = likelihood + log_probs.transpose(1, 2)
        tgt_prob = tgt_prob.logsumexp(dim=-1)
        tgt_char = torch.max(tgt_prob, dim=-1)[1]
        return tgt_char.unsqueeze(-1), likelihood
```
Also realize this isn't necessary as a tuple. Will remove with qol update.

Reply:
That's true, you're not changing the likelihood.